### PR TITLE
Like dislike

### PR DESCRIPTION
**Pull request by vijaytembugade**
- Added like and dis like functionality 
- Added Bookmarkig the post functionality

### CODE REVIEW

**Review by manojasarsa — APPROVED** [review]
The like and dislike functionality is working as expected.
Good to merge!

**Review by pritam-kr — APPROVED** [review]
The like and dislike functionality is working as expected. Good to merge.

File: src/App.js [hunk 1]
```diff
@@ -35,6 +36,14 @@ function App() {
               </PrivateRoutes>
             }
           />
+          <Route
+            path="/bookmark"
+            element={
+              <PrivateRoutes>
```
Comment (AbhishekkGautam) [inline src/App.js:42]: We can also wrap all the routes inside `<Route to="/" element={<PrivateRoute />}></Route>`

File: src/Components/Posts/Posts.js [hunk 2]
```diff
@@ -1,11 +1,28 @@
...
 import { Link } from "react-router-dom";
-import CommentModal from "../../Features/Comments/Components/CommentModal";
 import { Modal } from "../Modal/Modal";
+import CommentModal from "../../Features/Comments/Components/CommentModal";
+import { useDispatch, useSelector } from "react-redux";
+import { dislikePost, likePost } from "../../Features/Posts/Slice/postSlice";
+import { Loader } from "../Loader/Loader";
 
 export const Posts = ({ post }) => {
+  console.log(post);
```
Comment (AbhishekkGautam) [inline src/Components/Posts/Posts.js:11]: remove console.log

File: src/Pages/Bookmark.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+const Bookmark = () => {
+  const dispatch = useDispatch();
+  const { bookmarks } = useSelector((store) => store.users);
+  const { posts, status } = useSelector((store) => store.posts);
+
+  useEffect(() => {
+    dispatch(getAllPosts());
+  }, [dispatch]);
+
+  const bookmarkedPosts = posts.filter((post) => bookmarks.includes(post._id));
```
Comment (AbhishekkGautam) [inline src/Pages/Bookmark.js:16]: we can also get bookmark posts like this - 
```
bookmarkPosts
    .map(id => allPosts.find(post => post._id === id))
    .reverse();
```
if we apply the filter on the `posts` array, then the order of bookmark posts will be jumbled. 
In the above approach, we're mapping over `bookmarkId` array & then finding the relevant id in `allPosts`. 
and then reverse the array...in this way, the recently added bookmark post will be on the top.

**Review by AbhishekkGautam — APPROVED** [review]
Overall code is clean & well structured

File: src/Pages/Bookmark.js (hunk 3 above)
Comment (vijaytembugade) [reply src/Pages/Bookmark.js:16]: Yes. I will do the changes thank for mentioning it

**Pull request merged**